mobile/android/android-emulator-services.ts

```diff
@@ -169,7 +313,7 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 			var emulatorDevices: string[] = [];
 			var outputRaw = this.$childProcess.execFile(this.$staticConfig.adbFilePath, ['devices']).wait().split(os.EOL);
 			_.each(outputRaw, (device: string) => {
-				var rx = device.match(/^(emulator-\d+)\s+device$/);
+				var rx = options.geny ? device.match(/^(\d+\.\d+\.\d+\.\d+:\d+)\s+device$/) : device.match(/^(emulator-\d+)\s+device$/);
```

Comment:
if geny option is specified maybe we can shorten the regex: ^((\d+\.){3}\d+:\d+)\s+device$